dotcom-rendering/src/lib/getFrontsAdPositions.ts

```diff
@@ -9,10 +13,66 @@ type GroupedCounts = {
 	standard: number;
 };
 
+type AdCandidate = Pick<DCRCollectionType, 'collectionType'>;
+
+const getMerchHighPosition = (collectionCount: number): number =>
+	collectionCount >= 4 ? 2 : 0;
+
+// This happens on the recipes front, where the first container is a thrasher see: https://github.com/guardian/frontend/pull/20617
+const hasFirstContainerThrasher = (collectionType: string, index: number) =>
+	index === 0 && collectionType === 'fixed/thrasher';
+
+const hasAdjacentCommercialContainer = (
+	collectionIndex: number,
+	merchHighPosition: number,
+) => collectionIndex === merchHighPosition;
+
+const hasAdjacentThrasher = (index: number, collections: AdCandidate[]) =>
+	collections[index + 1]?.collectionType === 'fixed/thrasher';
```

Comment:
Small naming note, but it’s technically only if the thrasher follows, right? We’re not looking at `index - 1`

Reply:
That looks to be right. This is mobile fronts code so is pre-existing. Interestingly, there's a comment above the `getMobileAdPositions` function which states that a mobile ad should not be inserted between a regular container and a thrasher. So I assume that we used to avoid placing ads next to thrashers, until someone decided that we needed to place more ads!

I'll update the names of the functions and the comments to match the current behaviour.